File: src/main/java/online/yudream/yudreamskin/entity/Skin.java
package online.yudream.yudreamskin.entity;

import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.DocumentReference;

import java.time.LocalDateTime;
import java.util.Map;

@Document("tb_skin")
public class Skin {

    private String id;
    @DocumentReference(lazy = true)
    private User user;

    private String fileName;
    private String name;
    private Integer status;
    private Integer like;
    private Map<String,String> metadata;
    private String skinType;
    @CreatedDate
    private LocalDateTime createdAt;

    public Skin(){

    }
    public Skin(String name, String file,Map<String,String> metadata,Integer status, Integer like,String type){
        this.name = name;
        this.fileName = file;
        this.metadata = metadata;
        this.status = status;
        this.like = like;
        this.skinType = type;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public String getFileName() {
        return fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getStatus() {
        return status;
    }

    public void setStatus(Integer status) {
        this.status = status;
    }

    public Integer getLike() {
        return like;
    }

    public void setLike(Integer like) {
        this.like = like;
    }

    public Map<String, String> getMetadata() {
        return metadata;
    }

    public void setMetadata(Map<String, String> metadata) {
        this.metadata = metadata;
    }

    public String getSkinType() {
        return skinType;
    }

    public void setSkinType(String skinType) {
        this.skinType = skinType;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }
}

File: src/main/java/online/yudream/yudreamskin/entity/Skin.kt
package online.yudream.yudreamskin.entity

import org.springframework.data.annotation.CreatedDate
import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.data.mongodb.core.mapping.DocumentReference
import org.springframework.data.mongodb.core.mapping.MongoId
import java.io.Serializable
import java.time.LocalDateTime

@Document(collection = "tb_skin")
data class Skin (
    val name: String? = null,
    val fileName: String? = null,
    val metadata: Map<String, String>? =null,
    val status: Int = 0, //0: 公开 1: 私密
    val like: Int = 0,
    val skinType: String = "skin", //或cape
    @MongoId
    val id: String? = null,
): Serializable {
    @DocumentReference(lazy = true)
    val user: User? = null
    @CreatedDate
    var createdAt: LocalDateTime? = null
}